src/inmanta/server/agentmanager.py

```diff
@@ -178,6 +178,35 @@ def get_agent_client(self, tid: uuid.UUID, endpoint: str) -> Optional[ReturnClie
     async def stop(self) -> None:
         await super().stop()
 
+    @protocol.handle(methods_v2.pause_agent, env="tid")
+    async def pause_agent(self, env: data.Environment, name: str, paused: bool) -> None:
+        if paused:
+            await self._pause_agent(env, name)
+        else:
+            await self._unpause_agent(env, name)
+
+    async def _pause_agent(self, env: data.Environment, endpoint: str) -> None:
+        key = (env.id, endpoint)
+        async with self.session_lock:
+            await data.Agent.pause(env=env.id, endpoint=endpoint, paused=True)
+            live_session = self.tid_endpoint_to_session.get(key)
+            if live_session is not None:
+                del self.tid_endpoint_to_session[key]
+                await live_session.get_client().set_state(endpoint, False)
+
+    async def _unpause_agent(self, env: data.Environment, endpoint: str) -> None:
+        key = (env.id, endpoint)
+        async with self.session_lock:
+            await data.Agent.pause(env=env.id, endpoint=endpoint, paused=False)
+            live_session = self.tid_endpoint_to_session.get(key)
+            if live_session:
+                await live_session.get_client().set_state(endpoint, True)
+            else:
+                session = self._get_session_for(tid=env.id, endpoint=endpoint)
```

Comment:
We never pause session that is not live, why should we unpause it? This can use some comments.

Reply:
live_session is probably a bad name. It means: "The session of an agent that is unpaused and primary". So basically, the session to an agent that is ready to deploy something. So, the first part of the if-block is redundant actually.